Packet++/src/SSLHandshake.cpp

```diff
@@ -1228,6 +1228,19 @@ SSLHandshakeMessage* SSLHandshakeMessage::createHandshakeMessage(uint8_t* data,
 		return nullptr;
 
 	ssl_tls_handshake_layer* hsMsgHeader = (ssl_tls_handshake_layer*)data;
+
+	if (dataLen >= 16)
+	{
+		if (*(uint64_t*)data <= 0xFFFFFF || hsMsgHeader->length1 >= 1)
+			// possibly Encrypted Handshake Message
+			// used heuristic:
+			// - first 5 bytes of the handshake message are zeroes
+			// - or wrong message length is over 64K
+			// - handshake layer of more than 16 byte
+			// - or message type makes so sense (handled through the switch statement)
+			return new SSLUnknownMessage(data, dataLen, container);
```

Comment:
We currently don't have `SSLEncryptedMessage`. I'm contemplating if we need one... what do you think?
Or we can keep is `SSLUnknownMessage` for now...

Reply:
I would leave it at ```SSLUnknownMessage``` for now. The most important thing to me is that encrypted handshake messages are not mistakenly interpreted as a different handshake type. Our approach should fix that for the most part.